Helm-415: Docs direct user to create a Custom Action when creating a Dashboard but this feature no longer exists in Grafana v9 (#780)
OpenNMS/grafana-plugin: docs/modules/getting_started/pages/basic_walkthrough.adoc

. Click the *Enable* or *Enable plugin* button.
+
NOTE: The Disable option indicates that the plugin is already enabled.
You can also choose *Update* to download the latest version of the plugin via Grafana's plugin repository.

[[bw-ds-setup]]
== Create a datasource

The type of data that you can display in a dashboard depends on the datasource that you select.

Before you create a dashboard, you must set up the datasource (xref:datasources:performance_datasource.adoc[performance], xref:datasources:flow_datasource.adoc[flow], or xref:datasources:entity_datasource.adoc[entity]).

> For this walk-through, we will configure a performance datasource.
> Note that you can follow the same process to configure an entity or flow datasource.

Reviewer: ```suggestion
For this walk-through, we will configure a entity datasource.
After completing these steps, repeat the same process to configure a performance datasource, and optionally a flow datasource if you are collecting flow data.
```

You need to configure only the *URL*, *Basic Auth*, and *Basic Auth Details* (username and password) fields.
See the https://grafana.com/docs/grafana/latest/datasources/[official Grafana documentation] for information on configuring other fields.

Follow these steps to set up a datasource:

. Navigate to the home page of your Grafana instance.

> . In the left navigation menu, select menu:Administration[Data Sources].
> . Select *Add New Data Source*.
> . In the *Search* box, type *OpenNMS Performance*.
> . Select the *OpenNMS Performance* datasource.

Reviewer: ```suggestion
. In the left navigation menu, select menu:Administration[Data Sources].
. Select *Add New Data Source*.
. In the *Search* box, type *OpenNMS Entities*.
. Select the *OpenNMS Entities* datasource.
```

The settings dialog is displayed:
+

> image::gf-data-config2.png["OpenNMS Performance datasource settings dialog displaying sample settings.", 500]

Reviewer: ```suggestion
image::gf-data-config.png["OpenNMS Entities datasource settings dialog displaying sample settings.", 500]
```

Follow these steps to set up a datasource:

. Navigate to the home page of your Grafana instance.
. In the left navigation menu, select menu:Administration[Data ources].
. Select *Add Data Source*.
. In the *Search* box, type *OpenNMS Entities*.

> . Select the *OpenNMS Entities* datasource.

Reviewer: User will want to repeat the same steps with a Performance datasource, and optionally a Flow datasource.
Personally, I would suggest stepping through a Perf DS first, as that would become the default datasource

. In the left navigation menu, click *Dashboards*.
. Click menu:New[New Dashboard].
. Click *Add Visualization*.

> . Choose your data source (for example, *OpenNMS Performance*).

Reviewer: This must be Entities for an Alarm Table visualization.
```suggestion
. Choose your data source (for example, *OpenNMS Entities*).
```

Author: Thanks. This documentation was the first thing I wrote when I started at ONMS 4 years ago. I believe Jesse had suggested starting with Entities data source. We could create a ticket to change the walkthrough to be Performance data source focused if that makes more sense.

** (Optional) Configure the settings in the bottom half of the panel to customize the graph.
. Update the panel's *Name*.
. Click *Save* at the top-right of the panel.

== Use data links to provide alarm context

Data links in Grafana let you create a user-defined operation (for example, a link to a node detail page or a knowledge base article, links based on alarm severity, and so on) that might help users diagnose an alarm.

> Once created, a link to the specified data will appear when you right-click an alarm in an alarms.

Reviewer 2: Change to something like:

Once created, links will appear in the Alarm Table data which can be left-clicked. If you have configured more than one Data Link for the panel, left-clicking an item will cause a context menu to display which allows you to choose which link to follow.

** (Optional) Configure the settings in the bottom half of the panel to customize the graph.
. Update the panel's *Name*.
. Click *Save* at the top-right of the panel.

== Use data links to provide alarm context

Data links in Grafana let you create a user-defined operation (for example, a link to a node detail page or a knowledge base article, links based on alarm severity, and so on) that might help users diagnose an alarm.
Once created, a link to the specified data will appear when you right-click an alarm in an alarms table:

> image::gf-data-link.png["Alarms table with three alarms. The context menu is displayed, listing three defined options."]

Reviewer 2: This is an image of the Alarm Table OpenNMS context menu, but not the Grafana Data Link context menu.
